lib/ActivityStreams.js

```diff
@@ -118,31 +112,31 @@ ActivityStreams.prototype = {
       case am.type("TOP_FRECENT_SITES_REQUEST"):
         this._memoized.getTopFrecentSites(msg.data).then(links => {
           let processedLinks = this._previewProvider.processLinks(links);
-          this._previewProvider.saveNewLinks(processedLinks);
+          this._previewProvider.asyncSaveNewLinks(processedLinks);
           const cachedLinks = this._previewProvider.getCachedLinks(processedLinks);
           this.send(am.actions.Response("TOP_FRECENT_SITES_RESPONSE", cachedLinks, {append}), worker);
         });
         break;
       case am.type("RECENT_BOOKMARKS_REQUEST"):
         this._memoized.getRecentBookmarks(msg.data).then(links => {
           let processedLinks = this._previewProvider.processLinks(links);
-          this._previewProvider.saveNewLinks(processedLinks);
+          this._previewProvider.asyncSaveNewLinks(processedLinks);
```

Comment:
Probably a `then` which encloses whatever needs to follow sequentially.

And yes, I didn't fix it because you want to take it on, right?